src/prepack-node-environment.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import invariant from "./invariant.js";
+import { ExecutionContext } from "./realm.js";
+import Serializer from "./serializer/index.js";
+import { Completion } from "./completions.js";
+import { Value } from "./values";
+import construct_realm from "./construct_realm.js";
+import initializeGlobals from "./globals.js";
+import { getRealmOptions, getSerializerOptions } from "./options";
+import { InitializationError } from "./prepack-standalone";
+import initializeBootstrap from "./intrinsics/node/bootstrap.js";
+import initializeProcess from "./intrinsics/node/process.js";
+
+import type { Options } from "./options";
+import { defaultOptions } from "./options";
+
+declare var process: any;
+
+export function prepackNodeCLI(filename: string, options: Options = defaultOptions, callback: Function) {
+  let serialized;
+  try {
+    serialized = prepackNodeCLISync(filename, options);
+  } catch (err) {
+    callback(err);
+    return;
+  }
+  callback(null, serialized);
+}
+
+export function prepackNodeCLISync(filename: string, options: Options = defaultOptions) {
+  if (process.version !== "v7.9.0") {
```

Comment:
Not at the moment. The strategy is very fragile because I only model the minimal needed. Which is why this whole thing is so hacky. The ideal would be to have more of an automatic pass-through or making more things abstract.